### PR TITLE
Added push(...) and pop(...) for SIMD registers

### PR DESCRIPTION
**Pull request by lohika-denis-kotov**
# Description
Added push(...) and pop(...) for SIMD registers

### CODE REVIEW

File: src/cpu/x64/jit_generator.hpp [hunk 1]
```diff
@@ -182,6 +184,91 @@ class jit_generator : public Xbyak::CodeGenerator, public c_compatible {
...
+    }
+
+    inline void push(const std::vector<Xbyak::Xmm> &xmms) {
+        std::vector<std::function<void()>> deferred_movs{};
+        size_t offset = 0;
+        for (size_t i = 0; i < xmms.size(); ++i) {
+            const auto& xmm = xmms[i];
+            if (xmm.isXMM()) {
+                deferred_movs.emplace_back([&]() {
+                    uni_vmovdqu(ptr[rsp + offset], xmm);
```
Comment (avoskoboinyk-lohika) [inline src/cpu/x64/jit_generator.hpp:210]: Maybe, 'offset' should be captured by value here?

Comment (lohika-denis-kotov) [reply src/cpu/x64/jit_generator.hpp:210]: Fixed

**Review by avoskoboinyk-lohika — APPROVED** [review]

File: src/cpu/x64/jit_generator.hpp (hunk 1 above)
Comment (avoskoboinyk-lohika) [reply src/cpu/x64/jit_generator.hpp:210]: This one may resolved now (I don't have rights to resolve comments)

**Comment by lohika-denis-kotov** [comment]
@ceciliapeng2011 @dmitry-gorokhov @maxnick @nshchego Could you please review this PR or assign reviewers ?

**Comment by maxnick** [comment]
@lohika-denis-kotov , do we really need to place those changes in oneDNN fork? M.b. we can locate them in the new `jit_kernel_base` class?

**Comment by lohika-denis-kotov** [comment]
@maxnick
> @lohika-denis-kotov , do we really need to place those changes in oneDNN fork? M.b. we can locate them in the new `jit_kernel_base` class?

Yes, it could be located in new `jit_kernel_base`.
I just do not understand when we should push either in `openvino/oneDNN` or in `openvino` repo ?

**Comment by maxnick** [comment]
> @maxnick
> 
> > @lohika-denis-kotov , do we really need to place those changes in oneDNN fork? M.b. we can locate them in the new `jit_kernel_base` class?
> 
> Yes, it could be located in new `jit_kernel_base`. I just do not understand when we should push either in `openvino/oneDNN` or in `openvino` repo ?

The main idea is to keep in the fork as less changes as possible. Keeping less changes in the fork helps to easily move to new versions of oneDNN. The `uni_` instructions are kept in the fork by historical reasons. Thanks to your team, we will adopt our own jit kernel base classes where we can locate all the stuff we need to write our own jit kernels. So, in my understanding, until we change some oneDNN code, it is better to keep changes in the main repo. For example, if we used the push/pop instructions to modify oneDNN jit_kernels, then it would make sense to introduce the changes to the fork.

**Comment by lohika-denis-kotov** [comment]
@maxnick 
> > @maxnick
> > > @lohika-denis-kotov , do we really need to place those changes in oneDNN fork? M.b. we can locate them in the new `jit_kernel_base` class?
> > 
> > 
> > Yes, it could be located in new `jit_kernel_base`. I just do not understand when we should push either in `openvino/oneDNN` or in `openvino` repo ?
> 
> The main idea is to keep in the fork as less changes as possible. Keeping less changes in the fork helps to easily move to new versions of oneDNN. The `uni_` instructions are kept in the fork by historical reasons. Thanks to your team, we will adopt our own jit kernel base classes where we can locate all the stuff we need to write our own jit kernels. So, in my understanding, until we change some oneDNN code, it is better to keep changes in the main repo. For example, if we used the push/pop instructions to modify oneDNN jit_kernels, then it would make sense to introduce the changes to the fork.

I've already have a PR with `uni_*` instructions fixes that requires all this `push` and `pop` instructions.
Without them it would be harder to fix, see:
https://github.com/openvinotoolkit/oneDNN/pull/149

**Comment by lohika-denis-kotov** [comment]
@maxnick 
> The main idea is to keep in the fork as less changes as possible. Keeping less changes in the fork helps to easily move to new versions of oneDNN. ...

As for me, it is better to have fast cycle for update in `oneAPI/oneDNN`

**Comment by lohika-denis-kotov** [comment]
Closed in favor of https://github.com/openvinotoolkit/oneDNN/pull/149

**Pull request closed without merging**